### ISSUE TITLE
PDF caching

### ISSUE THREAD

**Issue by dev-zero** [indico/indico#3416]
In our Indico installation it seems that generated PDFs are not cached at all. When downloading for example a PDF from the contribution (`https://OURHOST/event/3/contributions/7/contribution.pdf`) , I get for each single request a new folder in Indico's `tmp` folder.
Furthermore, when using wget's mirror function `wget --mirror https://OURHOST/`, `wget` complains about a missing `Last-modified header missing -- time-stamps turned off`, which seems that fronting Indico with a caching proxy likely is not going to help since the proxy will not be able to determine whether a resource was updated.

**Comment by ThiefMaster** [comment]
Correct, they are not cached at the moment. Generating them is quite fast (we do cache e.g. the book of abstracts which may be much slower to build), and the tmp folder is regularly cleared by a task.

I'm not sure how relevant the usecase of a caching proxy in front of Indico is. We don't really support this, it would be pretty useless for logged-in users anyway.

Out of curiosity, why do you mirror your whole indico instance using wget? For a particular event you can already generate a static version from within Indico itself.

**Comment by ThiefMaster** [comment]
Ah, it's someone else doing it. In that case I would consider blocking their IP address on the webserver level. You could also look into using a custom robots.txt blocking wget, at least from the URLs generating PDFs; by default it honors the rules in there when using it to mirror a site.
